File: Python/Niveau 1/7 – Conditions avancées, opérateurs booléens/Casernes de pompiers.py
paires = int(input())
for i in range(paires):
   xmin1 = int(input())
   xmax1 = int(input())
   ymin1 = int(input())
   ymax1 = int(input())
   xmin2 = int(input())
   xmax2 = int(input())
   ymin2 = int(input())
   ymax2 = int(input())

   if (xmax2 <= xmin1 or xmax1 <= xmin2) or (ymax2 <= ymin1 or ymax1 <= ymin2):
      print("NON")
   else:
      print("OUI")

File: Pascal/Niveau 1/7 – Conditions avancées, opérateurs booléens/Casernes de pompiers.pas
program Solution;
var
   xMin1, xMin2, xMax1, xMax2, yMin1, yMin2, yMax1, yMax2, paires, i: LongInt;
begin
   read(paires);
   for i := 1 to paires do
   begin
      read(xMin1, xMax1, yMin1, yMax1, xMin2, xMax2, yMin2, yMax2);
      if ((xMax2 <= xMin1) or (xMax1 <= xMin2)) or ((yMax2 <= yMin1) or (yMax1 <= yMin2)) then
      begin
         writeln('NON');
      end
      else
      begin
         writeln('OUI');
      end;
   end;
end.